dds/DCPS/transport/rtps_udp/RtpsUdpReceiveStrategy.cpp

```diff
@@ -318,16 +327,25 @@ RtpsUdpReceiveStrategy::deliver_from_secure(const RTPS::Submessage& submessage)
     return;
 
   } else {
-    ACE_DEBUG((LM_WARNING, "(%P|%t) RtpsUdpReceiveStrategy: "
-               "preprocess_secure_submsg failed RPCH %d, [%d.%d]: %C\n",
+    ACE_DEBUG((LM_WARNING, ACE_TEXT("(%P|%t) RtpsUdpReceiveStrategy: ")
+               ACE_TEXT("preprocess_secure_submsg failed RPCH %d, [%d.%d]: %C\n"),
                peer_pch, ex.code, ex.minor_code, ex.message.in()));
     return;
   }
 
   if (!ok) {
-    ACE_DEBUG((LM_WARNING, "(%P|%t) RtpsUdpReceiveStrategy: "
-               "decode_datawriter/reader_submessage failed [%d.%d]: %C\n",
-               ex.code, ex.minor_code, ex.message.in()));
+    bool dw = category == DATAWRITER_SUBMESSAGE;
```

Comment:
Since the call to the plugin failed, this method shouldn't assume that `category` is initialized to something useful (8.5.1.9.6).

Reply:
I think my comment from last week here was wrong.  I thought it was dealing with the preprocess step, not the decode step.